### ISSUE TITLE
scatter plots with datetime x-axis and more than 1000 data points do not render all points

### ISSUE THREAD

**Issue by cprice404** [plotly/plotly.py#4298]
I believe this issue is similar to https://github.com/plotly/plotly_express/issues/145 .

I am using plotly 5.15.0 in jupyter lab 3.6.3.

My browsers (have tried firefox and chrome) support webgl.

When I render a scatter plot where the x-axis values are datetime objects,

if the plot has 1000 points, it renders fine:

```
size = 1000
x = np.random.randint(1690332167190, 1690332667190, size=size)
y = np.random.randint(0, 10000, size=size)
df = pd.DataFrame()
df['x'] = pd.to_datetime(x, unit='ms')
df['y'] = y
fig = px.scatter(df, x='x', y='y')
fig
```

![image](https://github.com/plotly/plotly.py/assets/1341071/70677fad-9a1d-42d3-9706-60df24000a50)

but if I change size to 1001, the render only shows a few clusters of values on the x-axis:

![image](https://github.com/plotly/plotly.py/assets/1341071/fc94bea9-66f8-4093-8f4a-3cc75fc3b8aa)

the hover/mouseover behavior makes it clear that there are a bunch of points that exist in the spaces between what is plotted on the graph:

![image](https://github.com/plotly/plotly.py/assets/1341071/fb0dadc2-afaf-4c12-abb9-b96cef802b92)

but the individual points don't render (or are rendered with the wrong x-value, not sure which).

**Comment by cprice404** [comment]
i presume this has to do with the switch from svg to webgl after 1000 data points, and if I add `render_mode=svg`, it works properly, but that isn't an ideal solution for graphs where we will have much more than 1000 data points.

**Comment by qchenevier** [comment]
It seems to be fixed with the release of v5.19.0: https://github.com/plotly/plotly.py/releases/tag/v5.19.0

**Comment by Coding-with-Adam** [comment]
I ran the code and it works for me as well. Closing issue.
